### PR TITLE
[Apache Tomcat] Add integration package with access data stream

### PR DESCRIPTION
**Pull request by niraj-elastic**
- Enhancement

## What does this PR do?

- Added 1 data stream (Access Logs).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added dashboards and visualizations.
- Added system test cases for the data stream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/apache_tomcat) directory.
- Run the following command to run tests. elastic-package test

## Screenshots
apache_tomcat-access-dashboard

## Meta issue
- #5200

## Related PRs
- #5424
- #6003
- #5926
- #5983

### CODE REVIEW

**Comment by agithomas** [comment]
@niraj-elastic , as there are many datastreams for tomcat, please create a meta (if not done already), link the datastreams to the meta.

**Comment by niraj-elastic** [comment]
>

File: packages/tomcat/img/tomcat-access-dashboard.png
Comment (harnish-elastic) [inline packages/tomcat/img/tomcat-access-dashboard.png:1]: For `distribution of access events by connection status` chart, can we disable truncate option for better readability?

**Comment by lalit-satapathy** [comment]
> > > @niraj-elastic , as there are many datastreams for tomcat, please create a meta (if not done already), link the datastreams to the meta.
> > 
> > 
> > Hey @agithomas There is already one [issue ](https://github.com/elastic/integrations/issues/5200) created for Tomcat, Would you like us to update this issue instead of creating new meta issue?
> 
> Niraj, you can have this as meta issue and update all the details here.

Yes.. Let's have all the details captured in the meta issue.

**Review by harnish-elastic — APPROVED** [review]
LGTM!

**Comment by ishleenk17** [comment]
@SubhrataK :  Please check if you have any reviews on the dashboards of Tomcat datastreams.

File: packages/tomcat/data_stream/access/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      ignore_missing: true
+  - grok:
+      field: event.original
+      tag: 'grok_parse_log'
+      patterns:
+        - '^(%{IP:source.ip}|%{DATA:source.user.name}) %{DATA:tomcat.access.http.ident} %{DATA:tomcat.access.http.useragent} \[%{DATA:_tmp.timestamp}\] \"%{DATA:http.request.method} %{DATA:url.original} HTTP\/%{DATA:http.version}\" %{NUMBER:http.response.status_code} %{DATA:destination.bytes} %{IP:tomcat.access.ip.local} %{DATA:_tmp.path} %{NOTSPACE:tomcat.access.connection_status}%{SPACE}%{NUMBER:tomcat.access.response_time} \"%{DATA:http.request.referrer}\" \"%{DATA:user_agent.original}\" X-Forwarded-For=\"%{DATA:_tmp.header_forwarder}(\")*$'
+        - '^(%{IP:source.ip}|%{DATA:source.user.name}) %{DATA:tomcat.access.http.ident} %{DATA:tomcat.access.http.useragent} \[%{DATA:_tmp.timestamp}\] \"%{DATA:http.request.method} %{DATA:url.original} HTTP\/%{DATA:http.version}\" %{NUMBER:http.response.status_code} %{DATA:destination.bytes} %{IP:tomcat.access.ip.local} %{DATA:_tmp.path} %{NOTSPACE:tomcat.access.connection_status} \"%{DATA:http.request.referrer}\" \"%{DATA:user_agent.original}\" X-Forwarded-For=\"%{DATA:_tmp.header_forwarder}(\")*$'
```
[the remark this answers is not in the document]
Comment (niraj-elastic) [reply packages/tomcat/data_stream/access/elasticsearch/ingest_pipeline/default.yml:28]: We can get this field from url.original, so we don't need to include _tmp.path field. I'll remove it.

[the remark this answers is not in the document]
Comment (ishleenk17) [reply packages/tomcat/data_stream/access/elasticsearch/ingest_pipeline/default.yml:28]: Are we able to extract all information in this field through the url original?
If yes, then we need not store this field.

[the remark this answers is not in the document]
Comment (niraj-elastic) [reply packages/tomcat/data_stream/access/elasticsearch/ingest_pipeline/default.yml]: Yes we are able to extract all the information.

File: packages/apache_tomcat/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        title: Collect logs from Apache Tomcat via file
+        description: Collect logs from Apache Tomcat instances using filestream input.
+icons:
+  - src: /img/logo.svg
+    title: Apache Tomcat logo
+    size: 32x32
+    type: image/svg+xml
+screenshots:
+  - src: /img/apache_tomcat-access-dashboard.png
+    title: Apache Tomcat cache dashboard
```
Comment (ishleenk17) [inline packages/apache_tomcat/manifest.yml:25]: Incorrect title

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Pull request merged**